lib/bolt/project.rb

```diff
@@ -167,5 +167,131 @@ def check_deprecated_file
         Bolt::Logger.deprecation_warning('Using project.yaml instead of bolt-project.yaml', msg)
       end
     end
+
+    def migrate_project(configured_inventory_path, outputter)
+      migrating_inv = configured_inventory_path || inventory_file
+      inventory_1_to_2(migrating_inv, outputter) if migrating_inv.file?
+      bolt_yaml_to_bolt_project(migrating_inv, outputter)
+    end
+
+    def backup_file(origin_path, outputter)
+      unless File.exist?(origin_path)
+        outputter.print_message "Could not find file #{origin_path}, skipping backup."
+        return
+      end
+
+      date = Time.new.strftime("%Y%m%d_%H%M%S%L")
+      backup_dir = File.join(path, '.bolt-bak')
+      FileUtils.mkdir_p(backup_dir)
+
+      filename = File.basename(origin_path)
+      backup_path = File.join(backup_dir, "#{filename}.#{date}.bak")
+
+      outputter.print_message "Backing up #{filename} from #{origin_path} to #{backup_path}"
+
+      # No matter which error we resuce first, rubocop complains the order is wrong
+      # rubocop:disable Lint/ShadowedException
+      begin
+        FileUtils.cp(origin_path, backup_path)
+      rescue Errno::EACCES, StandardError => e
+        raise Bolt::FileError.new("#{e.message}; unable to create backup of #{filename}.", origin_path)
+      end
+      # rubocop:enable Lint/ShadowedException
+    end
+
+    private def bolt_yaml_to_bolt_project(migrating_inv, outputter)
+      # If bolt-project.yaml already exists
+      if project_file?
+        outputter.print_message "bolt-project.yaml already exists in Bolt "\
+          "project at #{path}. Skipping project file update."
+
+      # If bolt.yaml doesn't exist
+      elsif !config_file.file?
+        outputter.print_message "Could not find bolt.yaml in project at "\
+          "#{path}. Skipping project file update."
+
+      else
+        config_data = Bolt::Util.read_optional_yaml_hash(config_file, 'config')
+        transport_data, project_data = config_data.partition do |k, _|
+          Bolt::Config::INVENTORY_OPTIONS.keys.include?(k)
+        end.map(&:to_h)
+
+        if transport_data.any?
+          outputter.print_message "Moving transportation configuration options "\
+            "'#{transport_data.keys}' from bolt.yaml to inventory.yaml"
```

Comment:
```suggestion
            "'#{transport_data.keys.join("', '")}' from bolt.yaml to inventory.yaml"
```

This currently shows the keys in a quoted array, which looks a bit weird:

```
Moving transportation configuration options '["ssh"]' from bolt.yaml to inventory.yaml
```